mypy/semanal.py

```diff
@@ -3686,6 +3691,11 @@ def visit_class_def(self, tdef: ClassDef) -> None:
         if tdef.info.mro:
             tdef.info.mro = []  # Force recomputation
             calculate_class_mro(tdef, self.fail_blocker)
+        if tdef.analyzed:
+            if isinstance(tdef.analyzed, TypedDictExpr):
+                self.analyze(tdef.analyzed.info.typeddict_type)
+            if isinstance(tdef.analyzed, NamedTupleExpr):
```

Comment:
We (intentionally) support only functional syntax for ``NewType``, therefore it can't appear in ``ClassDef`` analyzed.